### PR TITLE
Handle `find`, `count` and `delete` correctly when collection doesn't exist

### PR DESCRIPTION
**Pull request by rumyantseva**
**Important!** I had to recreate the PR as I had messed up everything in my previous one. Sorry for that!
 The previous one was #698.

## Description

This PR refs #651 (covers a part of it)

Issue #651 is quite a big one, so I suggest dividing it into two PRs.

**In the scope of this PR:**
- `find` should return an empty response, with no errors.
- `count` should return 0.
- `delete` shouldn't return any errors.

**In a separate issue and PR:**
- `findAndModify` (this method is tricky as it needs to create a doc that represents data from both `query` and `update` docs).
- `update` (same reason).

**Other fixes in the current PR:**
Refactored `TestCreateDrop` to use subtests

## Testing

- I added tests for `count`, `find`, and `delete` to cover cases when collection doesn't exist.
- In addition, I manually checked various cases to compare responses between "pg-based implementation" and "proxy" to make sure that the logic is the same.

## Checklist

- [x] Code is regenerated if needed (`task gen`).
- [x] Code is formatted (`task fmt`).
- [x] Docs are formatted (`task docs-fmt`).
- [x] Tests passed (`task test`).
- [x] Linters passed (`task lint`).
- [x] PR assignee is set.
- [x] PR labels are set.
- [x] Project and iternation are set.
- [x] Critical codecov complaints are considered.
- [x] PR title and body are in accordance with the Conform PR GitHub Action.
- [x] Other GitHub Actions passed.

### CODE REVIEW

File: internal/handlers/pg/pgdb/pool_test.go [hunk 1]
```diff
@@ -187,3 +187,82 @@ func TestConcurrentCreate(t *testing.T) {
 		}
 	}
 }
+
+func TestPool_TableExists(t *testing.T) {
```
Comment (AlekSi) [inline internal/handlers/pg/pgdb/pool_test.go:191]: We don't use underscores in test names. We also don't create tests for each function if several functions could be tested together. I think this test the and the following one could be combined together with several subtests

File: internal/handlers/pg/pgdb/pool_test.go [hunk 2]
```diff
@@ -187,3 +187,82 @@ func TestConcurrentCreate(t *testing.T) {
...
+	pool := testutil.Pool(ctx, t, nil, zaptest.NewLogger(t))
+
+	schemaName := testutil.SchemaName(t)
+	tableName := testutil.TableName(t)
+
+	t.Cleanup(func() {
+		pool.DropSchema(ctx, schemaName)
+	})
+
+	// Case 1: schema does not exist, table does not exist
```
Comment (AlekSi) [inline internal/handlers/pg/pgdb/pool_test.go:204]: We use subtests for that

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Review by seeforschauer — COMMENTED** [review]
Would you please add some comments about testing naming and grouping? Thank you.

File: internal/handlers/pg/pgdb/pool_test.go (hunk 1 above)
Comment (rumyantseva) [reply internal/handlers/pg/pgdb/pool_test.go]: Fixed

File: internal/handlers/pg/pgdb/pool_test.go (hunk 2 above)
Comment (rumyantseva) [reply internal/handlers/pg/pgdb/pool_test.go]: Fixed

File: internal/handlers/pg/pgdb/pool_test.go [hunk 3]
```diff
@@ -93,39 +93,107 @@ func TestCreateDrop(t *testing.T) {
...
+		err := pool.CreateSchema(ctx, schemaName)
+		require.NoError(t, err)
 
-	err = pool.CreateTable(ctx, schemaName, tableName)
-	require.Equal(t, pgdb.ErrAlreadyExist, err)
+		// Schema exists ->
+		// - schema creation is not possible
+		// - table drop is not possible
+		// - table creation is possible
+		// - schema drop is possible (only once)
```
Comment (seeforschauer) [inline internal/handlers/pg/pgdb/pool_test.go:142]: :+1:

File: internal/handlers/pg/pgdb/pool_test.go [hunk 4]
```diff
@@ -93,39 +93,107 @@ func TestCreateDrop(t *testing.T) {
...
+
+		t.Cleanup(func() {
+			pool.DropSchema(ctx, schemaName)
+		})
+
+		// Schema does not exist ->
+		// - table drop is not possible
+		// - schema drop is not possible
+		// - table creation is not possible
+		// - shema creation is possible
```
Comment (seeforschauer) [inline internal/handlers/pg/pgdb/pool_test.go:110]: :+1:

**Review by seeforschauer — DISMISSED** [review]

**Review by w84thesun — DISMISSED** [review]

**Review by AlekSi — APPROVED** [review]
:shipit:

**Pull request merged**